bridges/snowbridge/pallets/system/src/lib.rs

```diff
@@ -623,12 +623,59 @@ pub mod pallet {
 			location: Box<VersionedLocation>,
 			metadata: AssetMetadata,
 		) -> DispatchResultWithPostInfo {
-			ensure_root(origin)?;
+			let origin_location: Location = T::SiblingOrigin::ensure_origin(origin)?;
 
 			let location: Location =
 				(*location).try_into().map_err(|_| Error::<T>::UnsupportedLocationVersion)?;
 
-			Self::do_register_token(&location, metadata, PaysFee::<T>::No)?;
+			// Ensure that origin location is some consensus system on a sibling parachain
+			let (para_id, agent_id) = ensure_sibling::<T>(&origin_location)?;
```

Comment:
We may need to reanchor the `origin_location` in context of Ethereum. Same as what we do for the `token_location`.
https://github.com/alistair-singh/polkadot-sdk/blob/28109f9b7599aabe36448f24eda348f5f0f84e88/bridges/snowbridge/pallets/system/src/lib.rs#L785-L790

Btw: In outbound V2 we add `DescribeGlobalPrefix` into `AgentIdOf`.
https://github.com/yrong/polkadot-sdk/blob/bee50a993d335b9a8e6aa4eced62599345b86c34/bridges/snowbridge/primitives/core/src/location.rs#L32